### PR TITLE
DX: improve How zkApps Work content

### PR DESCRIPTION
**Pull request by barriebyron**
This PR is first pass at verifying the docs for How zkApps Work:
- Part of the technical audit for hard fork #466 
- General effort to improve the content #417 

- current doc page https://docs.minaprotocol.com/zkapps/how-zkapps-work
- preview of this updated doc https://docs2-git-417-improve-how-zkapps-work-minadocs.vercel.app/zkapps/how-zkapps-work 

This PR applies updates for consistency, readability, and style.

correct formatting errors (bug), remove captions (redundant), tighten language, apply consistent vocabulary, and update as required for accuracy and consumability

Feedback most welcome!

### CODE REVIEW

File: docs/zkapps/how-zkapps-work.mdx [hunk 1]
```diff
@@ -24,153 +24,139 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
...
+- The term _zkApp_ refers to the UI + the smart contract.
 
-### Zero-knowledge based smart contracts
+## Zero knowledge-based smart contracts
 
-Because zkApps are based on **zero-knowledge proofs (zk-SNARKs)**, a zkApp developer writes what is called a **“circuit”**. A circuit is the method from which a **prover function** and a corresponding **verifier function** are derived during the build process.
+zkApps are based on zero knowledge proofs (zk-SNARKs). As a zkApp developer, you use the zkApp CLI to scaffold and deploy your project. 
 
-The **prover function** is the function that executes a smart contract’s custom logic.
+A circuit is the method from which a prover function and a corresponding verifier function are derived during the build process.
```
Comment (barriebyron) [inline docs/zkapps/how-zkapps-work.mdx:47]: what is a circuit now? provable something?

Comment (Trivo25) [reply docs/zkapps/how-zkapps-work.mdx:47]: It's now `Provable code`. You could rephrase this paragraph to something like this

> Provable code is written by using SnarkyJS, it generated a prover function and a corresponding verifier function that are derived during the build process.

File: docs/zkapps/how-zkapps-work.mdx [hunk 2]
```diff
@@ -24,153 +24,139 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
...
 
-The **prover function** runs in a user’s web browser as part of the zkApp. When interacting with a zkApp's UI, users enter any data (e.g. “buy ABC for y price”) required as input to the prover function, which then generates a zero-knowledge proof.
+The prover function is the function that executes a smart contract's custom logic and runs in a user's web browser as part of the zkApp. When interacting with a zkApp UI, users enter any data (for example, buy ABC for y price) that is required as input to the prover function, which then generates a zero knowledge proof.
```
Comment (Trivo25) [inline docs/zkapps/how-zkapps-work.mdx:49]: probably worth mentioning that the prover function actually generated a proof of the executed code!

File: docs/zkapps/how-zkapps-work.mdx [hunk 3]
```diff
@@ -24,153 +24,139 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
...
-- Run your prover function to run your smart contract
-- Generate a verification key to deploy your smart contract
+From this file, you can:
 
-The **prover function** runs in a user’s web browser.
+- Run a prover function to run your smart contract
+- Generate a verification key to deploy your smart contract
 
-The **verification key** lives on chain for a given zkApp account and is used by the Mina network to verify that a zero-knowledge proof has met all constraints defined in the prover. A verification key is required in order to create a zkApp account. 
+While the prover function runs in a user's web browser, the verification key lives on-chain for a given zkApp account and is used by the Mina network to verify that a zero knowledge proof has met all constraints defined in the prover. A verification key is required to create a zkApp account.
```
Comment (Trivo25) [inline docs/zkapps/how-zkapps-work.mdx:76]: You can also use the verifier function (or verification key) to verify proofs off-chain, but yes the main usage for the verifier function is the Mina network

File: docs/zkapps/how-zkapps-work.mdx [hunk 4]
```diff
@@ -24,153 +24,139 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
...
 
-The next two sections go into more detail about how the prover and verifier functions are used when a zkApp is deployed and when users interact with a zkApp.
+## Deploy a smart contract
 
-### Deploying a smart contract
+Smart contracts are deployed to the Mina network using the Mina zkApp CLI. The deployment process sends a transaction that contains the verification key to an address on the Mina blockchain.
```
Comment (Trivo25) [inline docs/zkapps/how-zkapps-work.mdx:80]: probably worth mentioning that the verification key is essentially the verifier function (or an artifact of it)

File: docs/zkapps/how-zkapps-work.mdx [hunk 5]
```diff
@@ -24,153 +24,139 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
...
+A regular Mina account can receive any transactions. 
 
-:::info
-
-When you deploy to a new Mina address, the Mina Protocol charges a 1 MINA fee for account creation. This fee is unrelated to zkAapps and helps to prevent Sybil or denial of service attacks.
-
-:::note
-
-### Deploying a zkApp UI
+Because zkApps are Mina Protocol's smart contracts powered by zero knowledge proofs, a zkApp account can successfully receive only transactions that contain a proof that satisfies the verifier function. Any transactions that do not pass the verifier function are rejected by the Mina network.
```
Comment (Trivo25) [inline docs/zkapps/how-zkapps-work.mdx:90]: I don't think this is 100% correct - you can specify custom permissions on smart contract accounts and by doing so, you can also send transactions to it that are not authorized by a proof

**Review by Trivo25 — APPROVED** [review]
nice, good job! Just left a few comments

File: docs/zkapps/how-zkapps-work.mdx (hunk 1 above)
Comment (barriebyron) [inline docs/zkapps/how-zkapps-work.mdx:47]:
```suggestion
Provable code is written by using SnarkyJS and generates a prover function and a corresponding verifier function that are derived during the build process.
```

File: docs/zkapps/how-zkapps-work.mdx [hunk 6]
```diff
@@ -24,153 +24,139 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
...
-- Run your prover function to run your smart contract
-- Generate a verification key to deploy your smart contract
+From this file, you can:
 
-The **prover function** runs in a user’s web browser.
+- Run a prover function to run your smart contract
+- Generate a verification key to deploy your smart contract
 
-The **verification key** lives on chain for a given zkApp account and is used by the Mina network to verify that a zero-knowledge proof has met all constraints defined in the prover. A verification key is required in order to create a zkApp account. 
+While the prover function runs in a user's web browser, the verification key lives on-chain for a given zkApp account and is used by the Mina network to verify that a zero knowledge proof has met all constraints defined in the prover. A verification key is required to create a zkApp account.
```
Comment (barriebyron) [inline docs/zkapps/how-zkapps-work.mdx:76]:
```suggestion
While the prover function runs in a user's web browser, the verification key lives on-chain for a given zkApp account and is used by the Mina network to verify that a zero knowledge proof has met all constraints defined in the prover. A verification key is required to create a zkApp account.

You can also use the verifier function or verification key to verify proofs off-chain.
```

File: docs/zkapps/how-zkapps-work.mdx [hunk 7]
```diff
@@ -24,153 +24,141 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
...
+zkApps are based on zero knowledge proofs (zk-SNARKs). As a zkApp developer, you use the zkApp CLI to scaffold and deploy your project. 
 
-The **prover function** is the function that executes a smart contract’s custom logic.
+Provable code is written by using SnarkyJS and generates a prover function and a corresponding verifier function that are derived during the build process.
 
-The **prover function** runs in a user’s web browser as part of the zkApp. When interacting with a zkApp's UI, users enter any data (e.g. “buy ABC for y price”) required as input to the prover function, which then generates a zero-knowledge proof.
+The prover function is the function that executes a smart contract's custom logic and runs in a user's web browser as part of the zkApp. When interacting with a zkApp UI, users enter any data (for example, buy ABC for y price) that is required as input to the prover function, which then generates a zero knowledge proof.
```
Comment (barriebyron) [inline docs/zkapps/how-zkapps-work.mdx:49]:
```suggestion
The prover function is the function that executes a smart contract's custom logic and runs in a user's web browser as part of the zkApp. The prover function generates a proof of the executed code. 

When interacting with a zkApp UI, users enter any data (for example, buy ABC for y price) that is required as input to the prover function, which then generates a zero knowledge proof.
```

File: docs/zkapps/how-zkapps-work.mdx [hunk 8]
```diff
@@ -24,153 +24,143 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
...
+While the prover function runs in a user's web browser, the verification key lives on-chain for a given zkApp account and is used by the Mina network to verify that a zero knowledge proof has met all constraints defined in the prover. A verification key is required to create a zkApp account.
 
-The **verification key** lives on chain for a given zkApp account and is used by the Mina network to verify that a zero-knowledge proof has met all constraints defined in the prover. A verification key is required in order to create a zkApp account. 
+You can also use the verifier function or verification key to verify proofs off-chain.
 
-The next two sections go into more detail about how the prover and verifier functions are used when a zkApp is deployed and when users interact with a zkApp.
+## Deploy a smart contract
 
-### Deploying a smart contract
+Smart contracts are deployed to the Mina network using the Mina zkApp CLI. The deployment process sends a transaction that contains the verification key to an address on the Mina blockchain.
```
Comment (barriebyron) [inline docs/zkapps/how-zkapps-work.mdx:84]:
```suggestion
Smart contracts are deployed to the Mina network using the Mina zkApp CLI. The deployment process sends a transaction that contains the verification key to an address on the Mina blockchain.
```
```suggestion
Smart contracts are deployed to the Mina network using the Mina zkApp CLI. The deployment process sends a transaction that contains the verification key, an output of the verifier function, to an address on the Mina blockchain.
```

**Pull request merged**